src/CoreLib/network.fs

```diff
@@ -1340,6 +1341,12 @@ and [<AllowNullLiteral>] NetworkConnections() as x =
         if Utils.IsNotNull dirpath then
             FileTools.DirectoryInfoCreateIfNotExists (dirpath) |> ignore
 
+    /// Initialize authentication using a shared secret password 
+    member x.InitializeAuthentication (pwd : string) : unit =
+        let keyFile =  Path.Combine( [| DeploymentSettings.LocalFolder; "Keys";  Guid.NewGuid().ToString("D") |] )
+        let keyFilePwd = Guid.NewGuid().ToString("D")
+        x.InitializeAuthentication(pwd, keyFile, String.Empty)
```

Comment:
keyfilePwd cannot be generated, has to be passed in.  Plus it seems you are passing String.Empty, so perhaps it is best to just take out the line "let keyFilePwd = Guid.NewGuid().ToString("D") "